lib/v2/screens/dashboard/dashboard.dart

```diff
@@ -156,6 +157,9 @@ class _DashboardState extends State<Dashboard> {
 
   Future<void> refreshData() async {
     BlocProvider.of<RatesBloc>(context)..add(const FetchRates());
+
+    _walletHeaderKey.currentState?.reload();
```

Comment:
Why does not raise the **TokenBalancesBloc** to the level of the **dashboard screeen** thus avoiding using a key and define a reload method in WalletHeader?

`_walletHeaderKey.currentState?.reload()`; ❌

`BlocProvider.of<TokenBalancesBloc>(context)..add(const OnLoadTokenBalances()); `✔️

Reply:
Hmm I figured even if it is at the dashboard level, I still need to send the bloc a message, so it still needs to be above in the widget hierarchy?

Reply:
I would love to do that, but it seems in order for me to be able to update from the refreshData() method in dashboard, I would have to have the balance bloc provider above the level of dashboard, so I can access it through context, same way we access the RateBloc

    BlocProvider.of<RatesBloc>(context)..add(const FetchRates());

That's cool and all but moving it even above dashboard (app? main?) seemed not a trade-off worth making to me, just to save some code - it would be a convoluted way to make this code more elegant.

I think the way it is it's appropriate, and my trade-off would be towards correctness, where this should be in the widget tree, vs. elegance

Reply:
If there is a way to move it to dashboard, let me know!

Reply:
It would not need to be a global Bloc like rates. 

Simply as the first widget in dasboard use a BlocProvider with the TokenBalancesBloc  ;)

<img src="https://user-images.githubusercontent.com/42857405/123500307-ca76e400-d602-11eb-8728-7059905a6290.png" width="500">

Reply:
> That's cool and all but moving it even above dashboard (app? main?) seemed not a trade-off worth making to me, just to save some code - it would be a convoluted way to make this code more elegant.

It's already complicated, this screen is spaguetti is pretty much code 😅.

You have to see it from the side of having a well uniform app having block working where they should and not having to do hacks to rise a state or call to upper leves.

Reply:
Ok I guess I don't see what the benefit is here, other than not using a key. I guess not using a key is a benefit. 

Sure I am not opposed to this.